pkg/controller/elasticsearch/services/services.go

```diff
@@ -146,7 +146,7 @@ func ElasticsearchURL(es esv1.Elasticsearch, pods []corev1.Pod) string {
 	}
 	if schemeChange {
 		// switch to sending requests directly to a random pod instead of going through the service
-		randomPod := pods[rand.Intn(len(pods))]
+		randomPod := pods[rand.Intn(len(pods))] // nolint:gosec
```

Comment:
These I think are actually useful since there are places where we might want to make sure we use a csprng, and we made that mistake in the past IIRC